ulauncher/modes/extensions/extension_finder.py

```diff
@@ -1,11 +1,38 @@
 import os
 
 
-def find_extensions(ext_dir):
+def is_extension(directory):
     """
-    Yields `(extension_id, extension_path)` tuples found in a given extensions dir
+    Tells whether the argument is an extension directory
     """
-    if os.path.exists(ext_dir):
+    manifest = os.path.join(directory, "manifest.json")
+    manifest = os.path.realpath(manifest)
+    return os.path.isfile(manifest)
+
+
+def locate_extension(ext_id, ext_dirs, default=None, default_first=False):
```

Comment:
Unless I'm missing it, `default` is never given by any caller. Additionally, the parameter `default_first` makes it sound like the default given will be used first, when instead the first path in the configuration is used, and the default (if it were ever set) would be returned directly.